### DISCUSSION TITLE
Hiding ports

### DISCUSSION THREAD

**Discussion by stepazel** [Blazor-Diagrams/Blazor.Diagrams#212] [Q&A]
Hey,

I've been using this library for a project at work, everything is great, but ports have been bugging me a little - I want to hide the ports and have links point directly to nodes.

Current situation:
![image](https://user-images.githubusercontent.com/59888389/178998110-9460539f-d967-40bc-aae8-1236aea1d6bc.png)

I'm using the default stylesheet.
If i try to change the position of ports using css, links are sometimes rendered differently, as shown below.
![image](https://user-images.githubusercontent.com/59888389/178998661-5bacf906-05d8-4ef7-b94d-d2c8a435a8ef.png)

Is there a way to hide the ports and have links connect to the node directly?
Thanks for the answers!!

**Comment by zHaytam** [comment]
Hello, amazing work!

The way I see it, your app will require ports because of the way/location links go from/to. If you use portless links, links will get drawn from the centers, which is probably not what you want.

When you say "links are sometimes rendered differently", which links are you referring to please? Because I see them all valid, except the one in the bottom-left which became diagonal instead of orthogonal.

What's exactly bugging you with the ports? The fact that they are there and visible? Or because they somehow limit you?

**Reply by stepazel** [reply]
Hey, thanks for the quick reply! Sorry I'm answering so late.

Exactly, the links show a direction of data flow of some sorts and portless links didn't suit my case, as you said.

Yea, the in this picture, only the bottom-left one is rendering incorrectly. But the issue arises more often with increasing complexity in the diagram, for example here:
![image](https://user-images.githubusercontent.com/59888389/180400122-242a30a6-b849-4af6-b850-4291b4694ef6.png)
Ideally, it would be something like this:
![image](https://user-images.githubusercontent.com/59888389/180400496-465998ed-0a38-4ff0-8126-e68514a713db.png)

The ports aren't bugging me personally, but the requirement from our project leader is clear - to have a close copy of older version, as seen here:
![image](https://user-images.githubusercontent.com/59888389/180400757-ac3f69a3-0e95-4d7c-ad7a-0fccbc4b01ec.png)
And the ports are bugging them, unfortunately.

Again, thanks for your answers and your work, it's amazing. 😄

**Comment by zHaytam** [comment]
Hello!

1. I see now. You're using the orthogonal router, right? And when you change the ports, it somehow fails to calculate the route. This is a problem, I am really not happy with the orthogonal algorithm and it shows now...
2. I see another blocker, however. In the last screenshot, the links are separated and don't start/end at the same location (port), which I'm afraid isn't possible currently in the library.

We can maybe think of a simple orthogonal router for your use case, but you'll need to think about my second point. If I may ask, what did you use in your original/older version?

**Reply by stepazel** [reply]
1. Yes, exactly, it somehow fails, although the only thing I did was set the margin to a little different value.
Do you think it would be possible to edit the algorithm a little? Maybe I can also look into it? (Although I'm not a very much experienced developer lol)
2. I've discussed it with my colleagues and this fortunately isn't an issue, so no worries. :) 

The original version is a WPF app and the nodes and links are all inserted manually (absolute positions), so it was a long and painful process to add/edit new nodes and especially links. So your library is helping us a ton in this regard

**Reply by zHaytam** [reply]
Since I can't look at your code, it's very hard for me to point you to what you can edit in the algorithm. Do you think you can maybe create an example project that shows the issue with as little code as possible?

**Comment by joseph-boeing** [comment]
Was there ever a solution to hiding ports? I understand to use OrthogonalRoutes, I must have ports but would like to change their size or transparency. Is there a way to do that? PortMpdel.Size does not seem to change the visual effect of the Port.
